pkg/compact/planner.go

```diff
@@ -93,6 +93,10 @@ func (p *tsdbBasedPlanner) plan(noCompactMarked map[ulid.ULID]*metadata.NoCompac
 	return nil, nil
 }
 
+func (p *tsdbBasedPlanner) PlanWithPartition(ctx context.Context, metasByMinTime []*metadata.Meta, partitionID int, errChan chan error) ([]*metadata.Meta, error) {
```

Comment:
Now that we can inject extensions, do we still need this method?

Reply:
Good catch. I removed this func from interface but forgot to remove them here.